api/src/db/models/agency_models.py

```diff
@@ -41,6 +41,7 @@ class Agency(ApiSchemaTable, TimestampMixin):
     agency_id: Mapped[int] = mapped_column(BigInteger, primary_key=True)
 
     agency_name: Mapped[str]
+    top_agency_name: Mapped[str]
```

Comment:
There shouldn't be any changes at the DB level for this PR, Mike H's recent PR added a property for this already https://github.com/HHS/simpler-grants-gov/pull/2517/files#diff-9b2da6ff61feaaaccb33635caeb9059e185c4f6c9d8334989097779e48a243aa which this PR should use.